cove/lib/common.py

```diff
@@ -234,3 +234,98 @@ def get_schema_validation_errors(json_data, schema_url, schema_name, current_app
         unique_validator_key = [validator_type, message, path_no_number]
         validation_errors[json.dumps(unique_validator_key)].append(value)
     return dict(validation_errors)
+
+
+def _get_schema_deprecated_paths(schema_name, schema_url, obj=None, current_path=(), deprecated_paths=None):
+    '''Get a list of deprecated paths (as tuples) in a schema'''
+    if deprecated_paths is None:
+        deprecated_paths = []
+
+    if schema_url and schema_url.startswith("http"):
+        obj = jsonref.loads(requests.get(schema_url + schema_name).text)
+    elif schema_name:
+        with open(schema_url + schema_name) as schema_file:
+            obj = jsonref.load(schema_file)
+
+    for prop, value in obj['properties'].items():
+        if current_path:
+            path = current_path + (prop,)
+        else:
+            path = (prop,)
+
+        if "deprecated" in value and path not in deprecated_paths:
+                deprecated_paths.append(path)
```

Comment:
I notice we're not yet getting the reason for the depreaction from the schema (`value["deprecated"]["description"]`), or the `deprecatedVersion`.